src/Microsoft.Android.Runtime.NativeAOT/Android.Runtime.NativeAOT/TypeMapping.cs

```diff
@@ -0,0 +1,77 @@
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Hashing;
+using System.Runtime.InteropServices;
+using System.Text;
+using Android.Runtime;
+
+namespace Microsoft.Android.Runtime;
+
+internal static class TypeMapping
+{
+	private static Dictionary<Type, string[]> s_javaClassNames = new ();
+
+	internal static bool TryGetType (string javaClassName, [NotNullWhen (true)] out Type? type)
+	{
+		ulong hash = Hash (javaClassName);
+
+		// the hashes array is sorted and all the hashes are unique
```

Comment:
Are you going to check the uniqueness of the hashcodes at build time to guarantee this?

An alternative way to deal with this would be to allow non-unique hashcodes and check all candidates in the map. If you do that, you can change the hashcodes to 32-bit that makes the map 2x smaller and dealing with hashcodes faster, at the cost of a rare hash collision.